src/store/bankdata/getters.ts

```diff
@@ -1,21 +1,39 @@
 import { GetterTree } from 'vuex';
 import { BankAccount } from '@src/view_models/BankAccount';
 import { Validity } from '@src/view_models/Validity';
+import { looksLikeIban } from '@src/util/bank_account_number_helpers';
 
+// It's important to note here that !bankDataIsInvalid is not the same as bankDataIsValid and
+// vice versa. This is because we have states Validity.INCOMPLETE and Validity.RESTORED. If
+// the validity of one of the values is one of these states then both of these will return false.
 export const getters: GetterTree<BankAccount, any> = {
 	bankDataIsInvalid: function ( state: BankAccount ): boolean {
-		return state.validity.bankdata === Validity.INVALID;
+		if ( state.values.accountNumber === '' || looksLikeIban( state.values.accountNumber ) ) {
+			return state.validity.accountNumber === Validity.INVALID;
+		}
+
+		return state.validity.accountNumber === Validity.INVALID || state.validity.bankCode === Validity.INVALID;
 	},
 	bankDataIsValid: function ( state: BankAccount ): boolean {
-		return state.validity.bankdata === Validity.VALID;
+		if ( state.values.accountNumber === '' || looksLikeIban( state.values.accountNumber ) ) {
+			return state.validity.accountNumber === Validity.VALID;
+		}
+
+		return state.validity.accountNumber === Validity.VALID && state.validity.bankCode === Validity.VALID;
+	},
+	accountNumber: function ( state: BankAccount ): string {
+		return state.values.accountNumber;
+	},
+	bankCode: function ( state: BankAccount ): string {
+		return state.values.bankCode;
 	},
-	getBankName: function ( state: BankAccount ): string {
+	bankName: function ( state: BankAccount ): string {
 		return state.values.bankName;
 	},
-	getAccountId: function ( state: BankAccount ): string {
+	iban: function ( state: BankAccount ): string {
 		return state.values.iban;
 	},
-	getBankId: function ( state: BankAccount ): string {
+	bic: function ( state: BankAccount ): string {
```

Comment:
Since we now track all bank data as store values, we could drop the getters (which are just redirects to the values) and save a few characters and CPU cycles ...

Reply:
This is kind've a toss up between accessing them with:

`store.state.bankdata.values.bic`
or
`store.getters[ 'bankdata/bic' ]`

Neither of them autocomplete, so in this case I think I prefer the getter. What do you think?

Reply:
I prefer the state, because it has no redirection and will be more efficient. But it's not very important to me.